### ISSUE TITLE
valid-types cannot parse `@throws {@link SomeType}`

### ISSUE THREAD

**Issue by LukeAbby** [gajus/eslint-plugin-jsdoc#1380]
## Expected behavior
I expected no error.

## Actual behavior
I get an error saying it's invalid. I considered posting this issue on https://github.com/jsdoc-type-pratt-parser based upon the guidance in this issue format but I don't believe it's part of the concern of that library. My hypothesis is that it expects the type to be of the form `@throws SomeType` or `@throws {SomeType}` not `@throws {@link SomeType}` and so is giving the wrong part of the tag to the parser.

I'm using `jsdoc.configs["flat/recommended-typescript"]` but I do apologize if this is some TypeScript-specific setting I should have toggled. For evidence that this _is_ valid TypeScript syntax look at https://tsdoc.org/pages/tags/throws/

## ESLint Config

```json
{
    "jsdoc/valid-types": "error"
}
```

## ESLint sample

```js
/**
 * @throws {@link Error}
 */
declare function someFunction(): void;
```

## Environment

- Node version: v20.15.0
- ESLint version v9.26.0
- `eslint-plugin-jsdoc` version: ^50.6.11

**Comment by LukeAbby** [comment]
Upon reflection this is likely be a duplicate of my other issue #1381. I opened it as two separate issues because I thought I saw two cases: `@throws {@link SomeType}` failing and `@otherTag {@link SomeType | some descriptive text}` that might have underlying differences.

However as you can see in the final version of the issue `@otherTag {@link SomeType}` also triggers the issue, I don't even mention link pretty text anymore. It took some playing around to realize that though.
